addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/type/HomematicTypeGeneratorImpl.java

```diff
@@ -214,7 +214,7 @@ private ThingType createThingType(HmDevice device, List<ChannelGroupType> groupT
         properties.put(Thing.PROPERTY_MODEL_ID, device.getType());
 
         URI configDescriptionURI = getConfigDescriptionURI(device);
-        if (configDescriptionProvider.getConfigDescription(configDescriptionURI, null) == null) {
+        if (configDescriptionProvider.getConfigDescription(configDescriptionURI, Locale.getDefault()) == null) {
```

Comment:
Not sure whether this is the correct thing to do. `ConfigDescriptionProvider.getConfigDescription()` has a `NonNullByDefault` annotation (which also applies to the locale parameter), but all users of the locale parameter either had no annotation at all or even had a `Nullable` annotation (e.g. `ProfileTypeRegistry.getProfileTypes`)

@kaikreuzer Do you know what the correct direction of fixing this is?

Reply:
It is correct that the `ConfigDescriptionProvider` interface promises that the framework will always pass a locale when calling this method, so this annotation is good and helpful.

We have here now the (rare) situation that the caller is not the framework, but the binding itself. But as it isn't calling any `ConfigDescriptionProvider`, but a `HomematicConfigDescriptionProvider`, my suggestion is to declare that we know for this one that it is safe to also call it with a null value by adding:

```
public interface HomematicConfigDescriptionProvider extends ConfigDescriptionProvider {

    @Override
    @Nullable
    ConfigDescription getConfigDescription(URI uri, @Nullable Locale locale);
```

This should solve the issue, right?